### PR TITLE
Expand tree views by default

### PR DESCRIPTION
**Pull request by RMacfarlane**
Fixes https://github.com/Microsoft/vscode-pull-request-github/issues/248

### CODE REVIEW

**Comment by rebornix** [comment]
I remember that @sbatten was hacking around the same thing ;)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sbatten** [comment]
@rebornix luckily I didn't go for this one and instead did the new openDiff setting

File: src/view/prChangesTreeDataProvider.ts [hunk 1]
```diff
@@ -47,6 +50,9 @@ export class PullRequestChangesTreeDataProvider extends vscode.Disposable implem
 
 		this._localFileChanges = fileChanges;
 		this._onDidChangeTreeData.fire();
+
+		// Workaround to expand the tree view
+		this._view.reveal(this.getChildren()[0]);
```
Comment (rebornix) [inline src/view/prChangesTreeDataProvider.ts]: Did you get this to work properly? In #750 I did the same thing but if `getChildren` is called by us and there is no real tree item bind to it, VSCode doesn't know how to expand to that node.

File: src/view/prsTreeDataProvider.ts [hunk 2]
```diff
@@ -54,6 +54,11 @@ export class PullRequestsTreeDataProvider implements vscode.TreeDataProvider<Tre
 		return node ? this._onDidChangeTreeData.fire(node) : this._onDidChangeTreeData.fire();
 	}
 
+	reveal(): void {
```
Comment (rebornix) [inline src/view/prsTreeDataProvider.ts]: 🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View.

File: src/view/prChangesTreeDataProvider.ts (hunk 1 above)
Comment (RMacfarlane) [reply src/view/prChangesTreeDataProvider.ts]: oh no, I convinced myself this was working when it wasn't. I thought I had the view set to collapsed and then saw it expand on reload, but I must have always just had it open. Let me see if there's a way to store and reuse the result of getChildren for this

Comment (RMacfarlane) [reply src/view/prChangesTreeDataProvider.ts]: actually this seems like the problem:

```ts
		if (typeof this.dataProvider.getParent !== 'function') {
			return Promise.reject(new Error(`Required registered TreeDataProvider to implement 'getParent' method to access 'reveal' method`));
		}
```

I'll try adding a `getParent`

Comment (rebornix) [reply src/view/prChangesTreeDataProvider.ts]: 🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View

Comment (rebornix) [reply src/view/prChangesTreeDataProvider.ts]: 🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View

Comment (rebornix) [reply src/view/prChangesTreeDataProvider.ts]: 🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View
🤞 #750 introduced a similar logic and has cache for tree items in Pull Request Changes View

**Comment by RMacfarlane** [comment]
I'm closing this out since I think https://github.com/Microsoft/vscode/issues/68438 is required to make this work properly, and since defaulting to a separate viewlet fixes this issue for most users

**Pull request closed without merging**